components/Chips/examples/ChipsSizingExampleViewController.m

```diff
@@ -0,0 +1,81 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "ChipsExamplesSupplemental.h"
+
+#import "MaterialChips.h"
+
+@implementation ChipsSizingExampleViewController {
+  MDCChipView *_chipView;
+  UISlider *_widthSlider;
+  UISlider *_heightSlider;
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  self.view.backgroundColor = [UIColor whiteColor];
+
+  _chipView = [[MDCChipView alloc] init];
+  _chipView.titleLabel.text = @"Material";
+  _chipView.imageView.image = [self faceImage];
+  _chipView.accessoryView = [self deleteButton];
+  [self.view addSubview:_chipView];
+
+  CGSize chipSize = [_chipView sizeThatFits:self.view.bounds.size];
+  _chipView.frame = (CGRect) { CGPointMake(20, 20), chipSize };
+
+  _widthSlider = [[UISlider alloc] initWithFrame:CGRectZero];
+  _widthSlider.maximumValue = 200;
+  _widthSlider.value = _chipView.frame.size.width;
+  [_widthSlider addTarget:self
+                   action:@selector(widthSliderChanged:)
+         forControlEvents:UIControlEventValueChanged];
+  [self.view addSubview:_widthSlider];
+
+  _heightSlider = [[UISlider alloc] initWithFrame:CGRectZero];
+  _heightSlider.maximumValue = 100;
+  _heightSlider.value = _chipView.frame.size.height;
+  [_heightSlider addTarget:self
+                    action:@selector(heightSliderChanged:)
+          forControlEvents:UIControlEventValueChanged];
+  [self.view addSubview:_heightSlider];
+}
+
+- (void)viewWillLayoutSubviews {
+  [super viewWillLayoutSubviews];
+
+  CGSize sliderSize = [_widthSlider sizeThatFits:self.view.bounds.size];
+  _widthSlider.frame = CGRectMake(20, 140, self.view.bounds.size.width - 40, sliderSize.height);
+  _heightSlider.frame =
+      CGRectMake(20, 140 + sliderSize.height + 20, self.view.bounds.size.width - 40, sliderSize.height);
+}
+
+- (void)widthSliderChanged:(UISlider *)slider {
+  CGRect frame = _chipView.frame;
```

Comment:
Since the frame rect may be non-standard, setting the width may cause it to shift locations.
For (extreme) example, 
```objectivec
CGRect frame = _chipView.frame; // (95, 95, -80, -80) == (15, 15, 80, 80)
frame.size.width = slider.value; // (95, 95, 100, -80) == (95, 15, 100, 80)
_chipView.frame = frame; // Has now moved 80 points in the X-axis
```